tensorflow/core/kernels/adjust_hue_op_gpu.cu.cc

```diff
@@ -0,0 +1,147 @@
+  const float new_h = h * 6.0f;
+  const float chroma = v * s;
+  const float x = chroma * (1.0 - fabsf(fmodf(new_h, 2.0f) - 1.0f));
+  const float new_m = v - chroma;
+  const bool between_0_and_1 = new_h >= 0.0 && new_h < 1;
+  const bool between_1_and_2 = new_h >= 1.0 && new_h < 2;
+  const bool between_2_and_3 = new_h >= 2 && new_h < 3;
+  const bool between_3_and_4 = new_h >= 3 && new_h < 4;
+  const bool between_4_and_5 = new_h >= 4 && new_h < 5;
+  const bool between_5_and_6 = new_h >= 5 && new_h < 6;
+  tuple.r = chroma * (between_0_and_1 || between_5_and_6) +
+    x * (between_1_and_2 || between_4_and_5) + new_m;
+  tuple.g = chroma * (between_1_and_2 || between_2_and_3) +
+    x * (between_0_and_1 || between_3_and_4) + new_m;
+  tuple.b =  chroma * (between_3_and_4 || between_4_and_5) +
+    x * (between_2_and_3 || between_5_and_6) + new_m;
+  return tuple;
+}
+
+__global__ void adjust_hue_nhwc(const int number_elements,
+                                const float * const input, float * const output, const float * const hue_delta) {
+  const float delta = hue_delta[0];
+  // multiply by 3 since we're dealing with contiguous RGB bytes for each pixel (NHWC)
+  const int idx = (blockDim.x * blockIdx.x + threadIdx.x) * 3;
+  // bounds check
+  if (idx > number_elements) {
+    return;
+  }
+  const HsvTuple hsv = rgb2hsv_cuda(input[idx], input[idx+1], input[idx+2]);
+  // hue adjustment
+  float new_h = fmodf(hsv.h + delta, 1.0f);
+  if (new_h < 0.0f) {
+    new_h = fmodf(1.0f + new_h, 1.0f);
+  }
+  const RgbTuple rgb = hsv2rgb_cuda(new_h, hsv.s, hsv.v);
+  output[idx] = rgb.r;
+  output[idx + 1] = rgb.g;
+  output[idx + 2] = rgb.b;
+}
+} // namespace internal
+
+template <>
+class AdjustHueOp<GPUDevice> : public AdjustHueOpBase {
+public:
+explicit AdjustHueOp(OpKernelConstruction* const context)
+  : AdjustHueOpBase(context) {}
+
+void DoCompute(OpKernelContext* const context,
+  const ComputeOptions& options) override {
+  const Tensor* input = options.input;
+  const Tensor* delta = options.delta;
+  Tensor* const output = options.output;
+  const int64 number_elements = input->NumElements();
+  const GPUDevice &d = context->eigen_gpu_device();
+  const auto stream = d.stream();
+  OP_REQUIRES(context, stream, errors::Internal("No GPU stream available."));
+  if (number_elements > 0) {
+    const CudaLaunchConfig config = GetCudaLaunchConfig(number_elements, d);
+    const float * const input_data = input->flat<float>().data();
+    const float * const delta_h = delta->flat<float>().data();
+    float * const output_data = output->flat<float>().data();
+    const int threads_per_block = config.thread_per_block;
+    const int block_count = (number_elements + threads_per_block - 1) / threads_per_block;
+    internal::adjust_hue_nhwc<<<block_count, threads_per_block, 0, stream>>>(
+      number_elements, input_data, output_data, delta_h
+    );
+  }
+}
+};
+
+REGISTER_KERNEL_BUILDER(Name("AdjustHue").Device(DEVICE_GPU),
```

Comment:
@zheng-xq It worked with nvcc from CUDA 8, but yes, I haven't checked with older compilers. I can try doing it the way you suggested if that helps generalize the build.